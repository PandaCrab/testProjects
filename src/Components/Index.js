import { ShippingInfo } from './Forms/ShippingInfo';
import { BillingInfo } from './Forms/BillingInfo';
import { Payment } from './Forms/Payment';
import { OrderPlate } from './OrderPlate/OrderPlate';


export { ShippingInfo, BillingInfo, Payment, OrderPlate };
