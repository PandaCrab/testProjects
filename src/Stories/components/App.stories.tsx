import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import { ApolloProvider } from '@apollo/client';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import { store } from '../../redux/store';
import App from '../../App';
import { apolloClient } from '../..';

export default {
    title: 'App',
    component: App
} as ComponentMeta<typeof App>;

const Template: ComponentStory<typeof App> = () => (
    <ApolloProvider client={apolloClient}>
        <Provider store={store}>
            <BrowserRouter>
                <App />
            </BrowserRouter>
        </Provider>
    </ApolloProvider>
);

export const Header = Template.bind({});
