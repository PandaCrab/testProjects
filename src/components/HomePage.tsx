import React from 'react';
import { useNavigate } from 'react-router-dom';

import { 
    StartPage,
    NavigationButton
} from '../styles/HomePageStyles';

const HomePage = () => {
    const navigate = useNavigate();

    return (
        <>
            <StartPage>
                <h1>Hello World</h1>
                <NavigationButton onClick={() => navigate('/shopping')}>Start shopping</NavigationButton>
            </StartPage>
        </>
    );
};

export default HomePage;
