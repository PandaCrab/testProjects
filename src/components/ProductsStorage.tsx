import { useState } from 'react';
import { 
    SortingState,
    IntegratedSorting,
    RowDetailState
} from '@devexpress/dx-react-grid';
import {
  Grid,
  Table,
  TableHeaderRow,
  TableRowDetail,
} from '@devexpress/dx-react-grid-material-ui';
// import { useDispatch, useSelector } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { useQuery } from '@apollo/client';

// import { AppDispatch, RootState } from '../types';

// import { getProductsStorage } from '../redux/ducks/stuff';
import { Heading, NavigationButton, StartPage } from '../styles/HomePageStyles';
import { TAKE_PRODUCTS_FROM_STORAGE } from '../api';

interface graphQlData {
    id: string,
    name: string,
    price: number,
    quantity: number
}

const RowDetail = ({ row }: any) => (
    <div>
        {row.quantity > 0 ? 
        `This item can be send today` :
        `Check later, out of stock`
        }
    </div>
)

const ProductsStorage = () => {
    const { error, data } = useQuery(TAKE_PRODUCTS_FROM_STORAGE);
    const navigate = useNavigate();
    // const dispatch = useDispatch<AppDispatch>();
    // const productsStorage = useSelector((state: RootState) => state.order.productsStorage);

    // useEffect(() => { 
    //     dispatch(getProductsStorage())
    // // eslint-disable-next-line react-hooks/exhaustive-deps
    // }, []);

    const [columns] = useState([
        { name: 'id', title: 'Id' },
        { name: 'name', title: 'Name' },
        { name: 'price', title: 'Price' },
        { name: 'quantity', title: 'Quantity' }
    ]);

    const [tableColumnExtensions] = useState([
        { columnName: 'id', width: '10%' },
        { columnName: 'name', width: '50%' },
        { columnName: 'price', width: '10%' },
        { columnName: 'quantity', width: '15%' },
      ]);

    return (
        <StartPage>
            <Heading>
                <h1>Hello in storage</h1>  <NavigationButton onClick={() => navigate('/')} >Home</NavigationButton>
            </Heading>
            <Grid
                rows={data.productsStorage.map(({ id, name, price, quantity }: graphQlData) => ({
                    id,
                    name,
                    price,
                    quantity
                }))}
                columns={columns}
            >
                <SortingState
                    defaultSorting={[{ columnName: 'name', direction: 'asc' }]}
                />
                <IntegratedSorting />
                <RowDetailState />

                <Table columnExtensions={tableColumnExtensions} />
                <TableHeaderRow showSortingControls />
                <TableRowDetail 
                    contentComponent={RowDetail}
                />
            </Grid>
        </StartPage>
    );
};

export default ProductsStorage;
