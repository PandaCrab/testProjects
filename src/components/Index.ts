import ShippingInfo from './forms/ShippingInfo';
import BillingInfo from './forms/BillingInfo';
import Payment from './forms/Payment';
import OrderPlate from './orderPlate/OrderPlate';
import SuccessBuying from './forms/SuccessBuying';
import Navigate from '../Navigate';
import Stuff from './orderPlate/Stuff';
import StuffItems from './orderPlate/StuffItems';
import Loader from './Loader';
import CardToPrint from './forms/CardToPrint';
import DropdownAddresses from './forms/dropdownAddresses';
import CountriesSelect from './forms/CountriesSelect';
import HomePage from './HomePage';
import ProductsStorage from './ProductsStorage';
import SaleChart from './SaleChart';

export { ShippingInfo,
    BillingInfo, 
    Payment, 
    OrderPlate, 
    SuccessBuying, 
    Navigate,
    Stuff,
    StuffItems,
    Loader,
    CardToPrint,
    DropdownAddresses,
    CountriesSelect,
    HomePage,
    ProductsStorage,
    SaleChart
};
