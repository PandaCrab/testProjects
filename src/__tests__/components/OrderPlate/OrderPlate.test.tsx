import { mount, shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';

import { OrderPlate } from '../../../components/Index';

const initialState = {order: {stuff: [
        {
            "id": 1,
            "imgUrl": "https://guesseu.scene7.com/is/image/GuessEU/M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80&op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0",
            "name": "Check Print Shirt",
            "color": "Grey+red+black",
            "price": 110
        }
    ]
}}
const mockStore = configureStore();
let store: any;

describe ('<OrderPlate />', () => {

    it('should render correctly order plate component', () => {
        store = mockStore(initialState);
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <OrderPlate />
                </BrowserRouter>
            </Provider>
        );

        expect(component).toMatchSnapshot();
    });

    it('should rendered without stuff', () => {
        store = mockStore([]);
        const component = shallow(
            <Provider store={store}>
                <OrderPlate />
            </Provider>
        );

        expect(component).toBeDefined();
    });

});

describe('Open and close buttons', () => {
    store = mockStore(initialState);
    const component = mount(
        <Provider store={store}>
            <BrowserRouter>
                <OrderPlate />
            </BrowserRouter>
        </Provider>);

    it('should display or hide order on mobiles, when click', () => {
        component.find('button#open-close-btn').simulate('click');
        expect(component.find('viewOrder')).toBeTruthy();

        component.find('button#open-close-btn').simulate('click');
        expect(component.find('OrderInfo').prop('displayMobile')).toEqual(false);
    });

    it('should hide order on mobiles, when click', () => {
        component.find('#close-btn').simulate('click');

        expect(component).toBeDefined()
    });

});

