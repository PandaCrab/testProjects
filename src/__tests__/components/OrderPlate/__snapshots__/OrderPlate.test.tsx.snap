// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderPlate /> should render correctly order plate component 1`] = `
Array [
  <section
    className="sc-ksZaOG ibhyab"
  >
    <section>
      <svg
        className="sc-kLLXSd kBBKOh"
        fill="currentColor"
        height="1em"
        id="close-btn"
        onClick={[Function]}
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M354 671h58.9c4.7 0 9.2-2.1 12.3-5.7L512 561.8l86.8 103.5c3 3.6 7.5 5.7 12.3 5.7H670c6.8 0 10.5-7.9 6.1-13.1L553.8 512l122.4-145.9c4.4-5.2.7-13.1-6.1-13.1h-58.9c-4.7 0-9.2 2.1-12.3 5.7L512 462.2l-86.8-103.5c-3-3.6-7.5-5.7-12.3-5.7H354c-6.8 0-10.5 7.9-6.1 13.1L470.2 512 347.9 657.9A7.95 7.95 0 0 0 354 671z"
        />
        <path
          d="M880 112H144c-17.7 0-32 14.3-32 32v736c0 17.7 14.3 32 32 32h736c17.7 0 32-14.3 32-32V144c0-17.7-14.3-32-32-32zm-40 728H184V184h656v656z"
        />
      </svg>
      <section
        className="sc-hAZoDl cQWdqi"
      >
        <h6
          className="sc-fnykZs kPWXFA"
        >
          Order Summary
        </h6>
         
        <button
          className="sc-fEOsli ciOOuy"
        >
          edit order
        </button>
      </section>
      <div
        className="sc-bjUoiL gLLEmD"
      >
        <section
          className="sc-hHLeRK dPNuhk"
        >
          <img
            alt="Check Print Shirt"
            className="sc-dmRaPn zCVrD"
            src="https://guesseu.scene7.com/is/image/GuessEU/M63H24W7JF0-L302-ALTGHOST?wid=1500&fmt=jpeg&qlt=80&op_sharpen=0&op_usm=1.0,1.0,5,0&iccEmbed=0"
          />
          <section
            className="sc-kgflAQ dbCiuf"
          >
            <h6
              className="sc-fLlhyt clxgpt"
            >
              Check Print Shirt
            </h6>
            <p
              className="sc-bBrHrO bXHGEt"
            >
              Grey+red+black
            </p>
            <p
              className="sc-bBrHrO bXHGEt"
            >
              Quantity: 1
            </p>
          </section>
          <p
            className="sc-ivTmOn bpCsUh"
          >
            110
            $
          </p>
        </section>
      </div>
      <section
        className="sc-cxabCf lbMhQR"
      >
        <section
          className="sc-llJcti iyHPPe"
        >
          <p
            className="sc-gicCDI bjluxt"
          >
            Subtotal
          </p>
          <p
            className="sc-gicCDI bjluxt"
          >
            Shipping
          </p>
          <p
            className="sc-gicCDI bjluxt"
          >
            Taxes
          </p>
        </section>
        <section
          className="sc-iIPllB iIWbyc"
        >
          <p
            className="sc-gicCDI bjluxt"
          >
            $
            0
          </p>
          <p
            className="sc-gicCDI bjluxt"
          >
            $
            Free
          </p>
          <p
            className="sc-gicCDI bjluxt"
          >
            $
            12.12
          </p>
        </section>
      </section>
      <section
        className="sc-ezWOiH imiMFu"
      >
        <p
          className="sc-bZkfAO gehmlG"
        >
          Total
        </p>
        <p
          className="sc-bZkfAO gehmlG"
        >
          $
          12.12
        </p>
      </section>
    </section>
    <section>
      <section
        className="sc-ikZpkk ZNTXr"
      >
        <p
          className="sc-jIZahH jsQqEc"
        >
          All purchases are subject to our
          <a
            className="sc-himrzO jeWplj"
            href="/TermsAndConditions"
            onClick={[Function]}
          >
            Terms and Conditions
          </a>
        </p>
      </section>
    </section>
  </section>,
  <button
    className="sc-gXmSlM jELbHU"
    id="open-close-btn"
    onClick={[Function]}
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 16 16"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M5.757 1.071a.5.5 0 0 1 .172.686L3.383 6h9.234L10.07 1.757a.5.5 0 1 1 .858-.514L13.783 6H15.5a.5.5 0 0 1 .5.5v1a.5.5 0 0 1-.5.5H.5a.5.5 0 0 1-.5-.5v-1A.5.5 0 0 1 .5 6h1.717L5.07 1.243a.5.5 0 0 1 .686-.172zM3.394 15l-1.48-6h-.97l1.525 6.426a.75.75 0 0 0 .729.574h9.606a.75.75 0 0 0 .73-.574L15.056 9h-.972l-1.479 6h-9.21z"
      />
    </svg>
  </button>,
]
`;
